$NetBSD: patch-CMakeLists.txt,v 1.2 2026/09/22 18:39:33 tsutsui Exp $

- add missing NOJOYSTICK CMake optoin
- skip SDL2 and ffmpeg pkg-config check if they are explicitly disabled
  by CMake options
  (XXX needs more fixes in WIN32 case)

--- CMakeLists.txt.orig	2026-09-22 17:50:57.274434169 +0000
+++ CMakeLists.txt
@@ -134,6 +134,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 option(NOMONITOR "Disable debug monitor support" OFF)
 option(NOAVI "Disable AVI support" OFF)
+option(NOJOYSTICK "Disable joystick support" OFF)
 option(NOSOUND "Disable sound" OFF)
 set(DELAY_TIMER_MS_DEFAULT 1 CACHE STRING "Default timer resolution in milliseconds")
 option(REPLAYDEBUG_FRAME "Enable replay debug frame" OFF)
@@ -358,6 +359,10 @@ if(NOAVI)
     target_compile_definitions(PC6001VX PRIVATE NOAVI)
 endif()
 
+if(NOJOYSTICK)
+    target_compile_definitions(PC6001VX PRIVATE NOJOYSTICK)
+endif()
+
 if(NOSOUND)
     target_compile_definitions(PC6001VX PRIVATE NOSOUND)
 endif()
@@ -450,53 +455,57 @@ else()
             target_link_libraries(PC6001VX PRIVATE ${X11_LIBRARIES})
         endif()
 
-        pkg_check_modules(SDL2 QUIET sdl2)
+        if(NOT NOJOYSTIC)
+            pkg_check_modules(SDL2 QUIET sdl2)
 
-        if(SDL2_FOUND)
-            target_include_directories(PC6001VX PRIVATE ${SDL2_INCLUDE_DIRS})
-            target_link_libraries(PC6001VX PRIVATE ${SDL2_LIBRARIES})
-        else()
-            target_compile_definitions(PC6001VX PRIVATE NOJOYSTICK)
+            if(SDL2_FOUND)
+                target_include_directories(PC6001VX PRIVATE ${SDL2_INCLUDE_DIRS})
+                target_link_libraries(PC6001VX PRIVATE ${SDL2_LIBRARIES})
+            else()
+                target_compile_definitions(PC6001VX PRIVATE NOJOYSTICK)
+            endif()
         endif()
 
-        pkg_check_modules(LIBAVFORMAT QUIET libavformat)
-        pkg_check_modules(LIBAVCODEC QUIET libavcodec)
-        pkg_check_modules(LIBSWSCALE QUIET libswscale)
-        pkg_check_modules(LIBSWRESAMPLE QUIET libswresample)
-        pkg_check_modules(LIBAVUTIL QUIET libavutil)
-
-        if(LIBAVFORMAT_FOUND AND LIBAVCODEC_FOUND AND LIBSWSCALE_FOUND AND LIBSWRESAMPLE_FOUND AND LIBAVUTIL_FOUND)
-            set(PC6001VX_FFMPEG_LIBRARY_DIRS
-                ${LIBAVFORMAT_LIBRARY_DIRS}
-                ${LIBAVCODEC_LIBRARY_DIRS}
-                ${LIBSWSCALE_LIBRARY_DIRS}
-                ${LIBSWRESAMPLE_LIBRARY_DIRS}
-                ${LIBAVUTIL_LIBRARY_DIRS}
-            )
-            list(REMOVE_DUPLICATES PC6001VX_FFMPEG_LIBRARY_DIRS)
-
-            target_compile_definitions(PC6001VX PRIVATE __STDC_CONSTANT_MACROS __STDC_FORMAT_MACROS)
-            target_include_directories(PC6001VX PRIVATE
-                ${LIBAVFORMAT_INCLUDE_DIRS}
-                ${LIBAVCODEC_INCLUDE_DIRS}
-                ${LIBSWSCALE_INCLUDE_DIRS}
-                ${LIBSWRESAMPLE_INCLUDE_DIRS}
-                ${LIBAVUTIL_INCLUDE_DIRS}
-            )
-
-            if(PC6001VX_FFMPEG_LIBRARY_DIRS)
-                target_link_directories(PC6001VX PRIVATE ${PC6001VX_FFMPEG_LIBRARY_DIRS})
+        if(NOT NOAVI)
+            pkg_check_modules(LIBAVFORMAT QUIET libavformat)
+            pkg_check_modules(LIBAVCODEC QUIET libavcodec)
+            pkg_check_modules(LIBSWSCALE QUIET libswscale)
+            pkg_check_modules(LIBSWRESAMPLE QUIET libswresample)
+            pkg_check_modules(LIBAVUTIL QUIET libavutil)
+
+            if(LIBAVFORMAT_FOUND AND LIBAVCODEC_FOUND AND LIBSWSCALE_FOUND AND LIBSWRESAMPLE_FOUND AND LIBAVUTIL_FOUND)
+                set(PC6001VX_FFMPEG_LIBRARY_DIRS
+                    ${LIBAVFORMAT_LIBRARY_DIRS}
+                    ${LIBAVCODEC_LIBRARY_DIRS}
+                    ${LIBSWSCALE_LIBRARY_DIRS}
+                    ${LIBSWRESAMPLE_LIBRARY_DIRS}
+                    ${LIBAVUTIL_LIBRARY_DIRS}
+                )
+                list(REMOVE_DUPLICATES PC6001VX_FFMPEG_LIBRARY_DIRS)
+
+                target_compile_definitions(PC6001VX PRIVATE __STDC_CONSTANT_MACROS __STDC_FORMAT_MACROS)
+                target_include_directories(PC6001VX PRIVATE
+                    ${LIBAVFORMAT_INCLUDE_DIRS}
+                    ${LIBAVCODEC_INCLUDE_DIRS}
+                    ${LIBSWSCALE_INCLUDE_DIRS}
+                    ${LIBSWRESAMPLE_INCLUDE_DIRS}
+                    ${LIBAVUTIL_INCLUDE_DIRS}
+                )
+
+                if(PC6001VX_FFMPEG_LIBRARY_DIRS)
+                    target_link_directories(PC6001VX PRIVATE ${PC6001VX_FFMPEG_LIBRARY_DIRS})
+                endif()
+
+                target_link_libraries(PC6001VX PRIVATE
+                    ${LIBAVFORMAT_LIBRARIES}
+                    ${LIBAVCODEC_LIBRARIES}
+                    ${LIBSWSCALE_LIBRARIES}
+                    ${LIBSWRESAMPLE_LIBRARIES}
+                    ${LIBAVUTIL_LIBRARIES}
+                )
+            else()
+                target_compile_definitions(PC6001VX PRIVATE NOAVI)
             endif()
-
-            target_link_libraries(PC6001VX PRIVATE
-                ${LIBAVFORMAT_LIBRARIES}
-                ${LIBAVCODEC_LIBRARIES}
-                ${LIBSWSCALE_LIBRARIES}
-                ${LIBSWRESAMPLE_LIBRARIES}
-                ${LIBAVUTIL_LIBRARIES}
-            )
-        else()
-            target_compile_definitions(PC6001VX PRIVATE NOAVI)
         endif()
     else()
         target_compile_definitions(PC6001VX PRIVATE NOJOYSTICK NOAVI)
